libevmasm/Assembly.cpp

```diff
@@ -985,6 +986,65 @@ LinkerObject const& Assembly::assemble() const
 		return assembleEOF();
 }
 
+[[nodiscard]] bytes Assembly::assembleOperation(AssemblyItem const& _item) const
+{
+	// solidity::evmasm::Instructions underlying type is uint8_t
+	// TODO: Change to std::to_underlying since C++23
+	return {static_cast<uint8_t>(_item.instruction())};
+}
+
+[[nodiscard]] bytes Assembly::assemblePush(AssemblyItem const& _item) const
+{
+	bytes ret;
+	unsigned pushValueSize = numberEncodingSize(_item.data());
+	if (pushValueSize == 0 && !m_evmVersion.hasPush0())
+		pushValueSize = 1;
+
+	// solidity::evmasm::Instructions underlying type is uint8_t
+	// TODO: Change to std::to_underlying since C++23
+	ret.push_back(static_cast<uint8_t>(pushInstruction(pushValueSize)));
+	if (pushValueSize > 0)
+		appendBigEndian(ret, pushValueSize, _item.data());
+
+	return ret;
+}
+
+[[nodiscard]] std::pair<bytes, Assembly::LinkRef> Assembly::assemblePushLibraryAddress(AssemblyItem const& _item, size_t _pos) const
+{
+	return {
+		// solidity::evmasm::Instructions underlying type is uint8_t
+		// TODO: Change to std::to_underlying since C++23
+		bytes(1, static_cast<uint8_t>(Instruction::PUSH20)) + bytes(20),
+		{_pos + 1, m_libraries.at(_item.data())}
+	};
+}
+
+[[nodiscard]] bytes Assembly::assembleVerbatimBytecode(AssemblyItem const& item) const
+{
+	return item.verbatimData();
+}
+
+[[nodiscard]] bytes Assembly::assemblePushDeployTimeAddress() const
+{
+	// solidity::evmasm::Instructions underlying type is uint8_t
+	// TODO: Change to std::to_underlying since C++23
+	return bytes(1, static_cast<uint8_t>(Instruction::PUSH20)) + bytes(20);
+}
+
+[[nodiscard]] bytes Assembly::assembleTag(AssemblyItem const& _item, size_t _pos, bool _addJumpDest) const
+{
+	solRequire(_item.data() != 0, AssemblyException, "Invalid tag position.");
+	solRequire(_item.splitForeignPushTag().first == std::numeric_limits<size_t>::max(), AssemblyException, "Foreign tag.");
+	solRequire(_pos < 0xffffffffL, AssemblyException, "Tag too large.");
+	size_t tagId = static_cast<size_t>(_item.data());
+	solRequire(m_tagPositionsInBytecode[tagId] == std::numeric_limits<size_t>::max(), AssemblyException, "Duplicate tag position.");
+	m_tagPositionsInBytecode[tagId] = _pos;
```

Comment:
WTF, didn't even notice that we have that one defined as `mutable` :) I expected you to have to move it to `m_tagPositionsInBytecode`. But now I see that our `assemble()` is `const` too... Well, TBH I don't like how `m_tagPositionsInBytecode` works and I'd design it differently myself, but I want the PR merged, so probably no point messing with it now.